docs/learn/old/prompts.md

```diff
@@ -0,0 +1,219 @@
+# Prompts
+
+When working with LLMs, a surprising amount of work can go into determining the final text which gets passed into the model. Let's take a look at some of the tools Mirascope provides to make formatting your prompts as seamless as possible.
+
+## BasePrompt
+
+`BasePrompt` is a base class which works across all model providers. To create a prompt using `BasePrompt`:
+
+- Define a class which inherits `BasePrompt`
+- Set the prompt via the docstring and denote any template variable with curly braces
+- Type your template variables (and type them correctly! `BasePrompt` is built on Pydantic's `BaseModel` so incorrect typing will result in a `ValidationError`).
+
+```python
+from mirascope.core import BasePrompt
+
+class BookRecommendationPrompt(BasePrompt):
+    """Recommend a {genre} book."""
+    
+    genre: str
+```
+
+You can also set the prompt via the `prompt_template` decorator, which allows you to use a standard docstring for your class if you prefer:
+
+```python
+from mirascope.core import prompt_template
+
+@prompt_template("Recommend a {genre} book.")
+class BookRecommendationPrompt(BasePrompt):
+    """Now this is a normal docstring."""
+    
+    genre: str
+```
+
+To call the model with a `BasePrompt`, create an instance of your class and invoke the `run()` method with the appropriate provider’s configuration decorator. The `run()` function will create a decorated function covered in more detail in the next section[link to calls].
+
+For the configuration decorator, you are required to set the `model` argument. You can also set  `stream` for streams[link] `response_model` for extractions[link], and `tools` for function calling/tools[link]. All other functionality is supported in the provider’s native API, and the original parameters can be passed directly as keyword arguments into the decorator.
```

Comment:
is it worth mentioning call params, json mode if we're going through all the functionalities? Or at least mention there's more

Reply:
This is the old doc